api/chronos/permissions.py

```diff
@@ -16,3 +17,24 @@ def has_object_permission(self, request, view, obj):
         if isinstance(obj, ChronosSubmission):
             obj = obj.preprint
         return PreprintPublishedOrAdmin().has_object_permission(request, view, obj)
+
+
+class SubmissionAcceptedOrPublishedOrPreprintContributor(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if isinstance(obj, ChronosSubmission):
```

Comment:
is it possible to get here and not be a chronos submission?

Reply:
`check_object_permission` is called in the view after the view finds the `ChronosSubmission` instance. If the instance does not exist, the view will raise a `NotFound`. So it is not possible to get there if the instance is not a `ChronosSubmission` or the instance does not exist.